enterprise/backend/src/metabase_enterprise/query_field_validation/api.clj

```diff
@@ -27,27 +27,37 @@
 
 (defn- cards-with-inactive-fields
   [sort-column sort-direction offset limit]
-  (let [additional-joins      (condp = sort-column
+  (let [sort-dir-kw           (keyword sort-direction)
+        additional-joins      (condp = sort-column
                                 "name"           []
-                                "collection"     [[(t2/table-name :model/Collection) :coll] [:= :coll.id :c.collection_id]]
+                                "collection"     [[(t2/table-name :model/Collection) :coll] [:or
+                                                                                             [:= :coll.id :c.collection_id]
+                                                                                             [:= nil :c.collection_id]]]
                                 "created_by"     [[(t2/table-name :model/User) :u] [:= :u.id :c.creator_id]]
```

Comment:
Why are these calls to `table-name` necessary? If they are, can you leave a comment explaining why?

Reply:
Since I'm using the model kw instead of the table name, I need `table-name`. It's very tempting to use `:core_user` instead of `(t2/table-name :model/User)` but my way feels more semantic somehow. (And will help us if we ever manage to rename `core_user` to `user`

Reply:
Can't you just use the model keyword though? Toucan2 should do the table name look up for you.

Reply:
I don't think so; afaik that only works when it's a specific param to the function, not in the query map as a `{:from ...}`

```
metabase.models.user> (t2/select-one :model/User {:select [:u.*]
                                                  :from [[:model/User] :u]})
Execution error (PSQLException) at org.postgresql.core.v3.QueryExecutorImpl/receiveErrorResponse (QueryExecutorImpl.java:2725).
ERROR: relation "model.User" does not exist
  Position: 19
  ```

Reply:
That's a pity